packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +189,176 @@ class Tooltip extends mixin(
+        //      moved to the tooltip. I'm assuming this is due to the debounce. Now that it's triggered on "click"
+        //      for keyboard users do we still want the debounce?
+        const isHidden = state !== 'shown';
+        this.tooltip.element.setAttribute('aria-hidden', isHidden.toString());
+        this.element.setAttribute('aria-expanded', (!isHidden).toString());
+
+        if (this._handleFocusinListener) {
+          this._handleFocusinListener = this.unmanage(
+            this._handleFocusinListener
+          ).release();
+        }
+
+        if (this._handleClickListener) {
+          this._handleClickListener = this.unmanage(
+            this._handleClickListener
+          ).release();
+        }
+
+        if (this._handleKeydownListener) {
+          this._handleKeydownListener = this.unmanage(
+            this._handleKeydownListener
+          ).release();
+        }
+
+        if (state === 'shown') {
+          this._hookCloseActions(this.element);
+          const focusableNode = this.tooltip.element.querySelector(
+            settings.selectorTabbable
+          );
+          if (focusableNode) {
+            focusableNode.focus();
+          } else {
+            this.tooltip.element.focus();
+            const tooltipTabindex = this.tooltip.element.getAttribute(
+              'tabindex'
+            );
+            if (__DEV__) {
+              warning(
+                !tooltipTabindex || tooltipTabindex === '-1',
+                'Tooltips without interactive elements must include tabindex="0" on the tooltip element.'
+              );
+            }
+          }
+
+          const hasFocusin =
+            'onfocusin' in this.element.ownerDocument.defaultView;
+          const focusinEventName = hasFocusin ? 'focusin' : 'focus';
+          this._handleFocusinListener = this.manage(
+            on(
+              this.element.ownerDocument,
+              focusinEventName,
+              this._handleFocusin,
+              !hasFocusin
+            )
+          );
+        } else {
+          this.element.focus();
+        }
+
+        if (typeof callback === 'function') callback();
+      }
     );
   }
 
   /**
-   * Attaches event handlers to show/hide the tooltip.
+   * Attaches event handlers to show the tooltip.
    * @param {Element} element The element to attach the events to.
    * @private
    */
   _hookOn(element) {
-    const hasFocusin = 'onfocusin' in window;
-    const focusinEventName = hasFocusin ? 'focusin' : 'focus';
-    [focusinEventName, 'blur', 'touchleave', 'touchcancel'].forEach(name => {
+    /**
+     * Setup the _debouncedHandleClick function for displaying a tooltip
+     * @param {Event} evt - user initiated event
+     * @param {Integer[]} [allowedKeys] - allowed key codes the user may press to open the tooltip
+     * @private
+     */
+    const setupDebounceClick = (evt, allowedKeys) => {
```

Comment:
Do you have some explanation wrt the series of changes to "triggering open" logic here...?

Reply:
Tooltip no longer has events tied to focus and blur but uses `click` (`focus + space or enter` for keyboard users).

WRT `setupDebounceClick` function: that is to support triggers that are semantic and non-semantic buttons to support backwards capability. Non-semantic buttons need to have JavaScript to manually add watching for space and enter were with the semantic button we can rely on the browser's default behavior.

WRT `on(element.ownerDocument, 'click',`: since the component is no longer tied to focus and blur events for triggering the open and closing of the tooltip being able to close the tooltip by clicking outside of it was no longer there.

Reply:
Thank you for clarifying @elizabethsjudd!

> WRT `setupDebounceClick` function: that is to support triggers that are semantic and non-semantic buttons to support backwards capability. Non-semantic buttons need to have JavaScript to manually add watching for space and enter were with the semantic button we can rely on the browser's default behavior.

Wondering if my reply to your "why debounce?" question helps.

> WRT `on(element.ownerDocument, 'click',`: since the component is no longer tied to focus and blur events for triggering the open and closing of the tooltip being able to close the tooltip by clicking outside of it was no longer there.

I believe [the existing "close-on-click-outside" logic](https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/components/src/components/floating-menu/floating-menu.js#L141-L156) still works, even if we no longer rely on `focus` event on triggering open - Given "clicking-outside" user gesture fires `blur` event. Please let me know if you see it's not the case.

Reply:
@asudoh When I removed the "focus/blur" and without my current change to add the 'click' event, the `handleBlur` referenced in your link did not handle the scenario when the user clicked outside of the tooltip to close it.